File: Java/Tema1_Ejercicio6.java
/*
 * TEMA 1: INTRODUCCIÓN BÁSICA - EJERCICIO 6
 * ==========================================
 * Escriba un programa que lea la base y altura de un triángulo 
 * y calcule su área.
 * 
 * EXPLICACIÓN:
 * - Área del triángulo: (base × altura) / 2
 */

import java.util.Scanner;

public class Tema1_Ejercicio6 {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        
        System.out.print("Digite la base del triángulo: ");
        double base = scanner.nextDouble();
        
        System.out.print("Digite la altura del triángulo: ");
        double altura = scanner.nextDouble();
        
        // Calcular área
        double area = (base * altura) / 2;
        
        System.out.println("\n=== TRIÁNGULO ===");
        System.out.println("Base: " + base);
        System.out.println("Altura: " + altura);
        System.out.printf("Área: %.2f%n", area);
        
        scanner.close();
    }
}

/*
 * SALIDA ESPERADA:
 * ===============
 * Digite la base del triángulo: 10
 * Digite la altura del triángulo: 5
 * 
 * === TRIÁNGULO ===
 * Base: 10.0
 * Altura: 5.0
 * Área: 25.00
 */

File: JS/Tema1_Ejercicio6.js
/*
 * TEMA 1: INTRODUCCIÓN BÁSICA - EJERCICIO 6
 * ===========================================
 * Calcular el área de un triángulo.
 * 
 * EXPLICACIÓN:
 * - Fórmula de Heron para área de triángulo
 * - Semiperímetro
 * - Validación de triángulo válido
 */

function calcularAreaTriangulo(a, b, c) {
    // Validar que sea un triángulo válido
    if (a + b <= c || b + c <= a || a + c <= b) {
        return "No es un triángulo válido";
    }
    
    // Fórmula de Heron
    const s = (a + b + c) / 2;
    const area = Math.sqrt(s * (s - a) * (s - b) * (s - c));
    return area;
}

const lado1 = 5, lado2 = 6, lado3 = 7;

console.log("=== ÁREA DE TRIÁNGULO ===");
console.log(`Lados: ${lado1}, ${lado2}, ${lado3}`);

const area = calcularAreaTriangulo(lado1, lado2, lado3);
if (typeof area === 'number') {
    console.log(`Área: ${area.toFixed(2)} unidades²`);
} else {
    console.log(area);
}

/*
 * SALIDA ESPERADA:
 * === ÁREA DE TRIÁNGULO ===
 * Lados: 5, 6, 7
 * Área: 14.70 unidades²
 */
